app/controllers/api/transformation_mappings_controller.rb

```diff
@@ -22,6 +22,19 @@ def validate_vms_resource(type, id, data = {})
       end
     end
 
+    def vm_flavor_fit_resource(_type, _id, data)
+      data["mappings"].collect do |mapping|
+        source      = Api::Utils.resource_search_by_href_slug(mapping["source_href"])
+        destination = Api::Utils.resource_search_by_href_slug(mapping["destination_href"])
```

Comment:
hmm, the above 2 seems to be confusing to be.  *_href implies an API href, i.e. http://localhost:3000/api/vms/1 and not the slug, which is normally named href_slug  (for vms/1).

If those are expected to be href_slugs, you can rename then accordingly, i.e. mapping["source_href_slug"] and mapping["destination_href_slug"].  

Optionally, you can take an object reference which can take either, so mapping["source"] and mapping["destination"] which identify the object as desired

``` 
{ "href" : "http://localhost:3000/api/vms/1" }
```

or

```
{ "href_slug" : "vms/1" }
```

Reply:
Do you have an example of the last alternative you mentioned (object reference)?

Reply:
look at http://manageiq.org/docs/reference/latest/api/reference/users#creating-users for group, it's just a hash which can be one of many identifiers, id, href, description, etc.  or zone reference for creating a provider http://manageiq.org/docs/reference/latest/api/reference/providers#creating-providers, same idea.  you can look at any of the helper methods for those parse_<something>. In this case you'd add support for href_slug in addition to href for example.